pkg/cloudprovider/cloudapi/azure/azure_compute.go

```diff
@@ -39,18 +40,32 @@ type computeServiceConfig struct {
 	vnetAPIClient          azureVirtualNetworksWrapper
 	resourceGraphAPIClient azureResourceGraphWrapper
 	resourcesCache         *internal.CloudServiceResourcesCache
-	inventoryStats         *internal.CloudServiceStats
-	credentials            *azureAccountConfig
-	computeFilters         map[string][]*string
+	// applicable only for vm resources where key represents selector.
+	vmResourceCache map[string]*internal.CloudServiceResourcesCache
+	inventoryStats  *internal.CloudServiceStats
+	credentials     *azureAccountConfig
+	selectorIndexer cache.Indexer
+}
+
+const selectorChanged = "selectorChanged"
+
+// selector struct is meant for storing details about CES configuration, created query string, if config got updated etc.
+type selector struct {
+	configChanged          bool
+	selectorNamespacedName string
+	filters                []*string
 	// selectors required for updating resource filters on account config update.
-	selectors map[string]*crdv1alpha1.CloudEntitySelector
+	ces *crdv1alpha1.CloudEntitySelector
 }
 
-type computeResourcesCacheSnapshot struct {
```

Comment:
keep it computeResourceCacheSnapot only.. they will be account level and add selector map in that.. dont put it in config